image/verify/src/verifier.rs

```diff
@@ -455,10 +467,37 @@ impl<Env: ImageVerificationEnv> ImageVerifier<Env> {
             raise_err!(FmcDigestMismatch)
         }
 
-        // TODO: Perform following Address checks
-        // 1. Load address is a valid ICCM Address
-        // 2. Entry Point is a valid ICCM Address
-        // 3. Entry Point is within the image
+        // TODO: Perform following Address check
+        // Entry Point is within the image
+        if !self.env.iccm_range().contains(&verify_info.load_addr) {
+            raise_err!(FmcLoadAddrInvalid)
+        }
+        if verify_info.load_addr % 4 != 0 {
+            raise_err!(FmcLoadAddrUnaligned)
+        }
+
+        if !self.env.iccm_range().contains(&verify_info.entry_point) {
+            raise_err!(FmcEntryPointInvalid)
+        }
+        if verify_info.entry_point % 4 != 0 {
+            raise_err!(FmcEntryPointUnaligned)
+        }
+
+        if self.env.dev_lifecycle(image) != Lifecycle::Unprovisioned
+            || !self.env.anti_rollback_disable(image)
```

Comment:
Is this logic right? "Check the SVN if we're provisioned OR anti-rollback is enabled" 

Shouldn't that be AND?

Reply:
Udpated